packages/lib-classifier/src/store/FeedbackStore.js

```diff
@@ -0,0 +1,78 @@
+import { autorun } from 'mobx'
+import { addDisposer, getRoot, onAction, types } from 'mobx-state-tree'
+
+import * as helpers from './feedback/helpers'
+import strategies from './feedback/strategies'
+
+const FeedbackStore = types
+  .model('FeedbackStore', {
+    isActive: types.optional(types.boolean, false),
+    rules: types.map(types.frozen({}))
+  })
+  .actions(self => {
+    function afterAttach () {
+      createSubjectObserver()
+      createClassificationObserver()
+    }
+
+    function createSubjectObserver () {
+      const subjectDisposer = autorun(() => {
+        const subject = getRoot(self).subjects.active
+        if (subject) {
+          self.reset()
+          self.createFeedbackRules(subject)
+        }
+      })
+      addDisposer(self, subjectDisposer)
+    }
+
+    function createClassificationObserver () {
+      const classificationDisposer = autorun(() => {
+        onAction(getRoot(self), (call) => {
```

Comment:
Why `getRoot(self)` here? That listens to every action on every store. Isn't it safer to only listen to actions on the classifications store?

Reply:
I think so.
We should consider changing here too: https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/store/WorkflowStepStore.js#L72